spec/reducers/swap.spec.js

```diff
@@ -0,0 +1,184 @@
+  });
+
+  it('should handle SWAP_ORIGIN_KIND', () => {
+    const newOriginKind = 'ETH';
+    const newDestinationKind = buildDestinationKind(
+      newOriginKind,
+      INITIAL_STATE.destinationKind
+    );
+    const fakeBityRates = {
+      BTCETH: 10,
+      ETHBTC: 0.01
+    };
+    expect(swap(undefined, swapActions.originKindSwap(newOriginKind))).toEqual({
+      ...INITIAL_STATE,
+      originKind: newOriginKind,
+      destinationKind: newDestinationKind,
+      destinationKindOptions: without(ALL_CRYPTO_KIND_OPTIONS, newOriginKind),
+      destinationAmount: buildDestinationAmount(
+        INITIAL_STATE.originAmount,
+        newOriginKind,
+        newDestinationKind,
+        fakeBityRates
+      )
+    });
+  });
+
+  it('should handle SWAP_DESTINATION_KIND', () => {
+    const newDestinationKind = 'REP';
+    const newOriginKind = buildOriginKind(
+      INITIAL_STATE.originKind,
+      newDestinationKind
+    );
+    const fakeBityRates = {
+      BTCETH: 10,
+      ETHBTC: 0.01
+    };
+    expect(
+      swap(undefined, swapActions.destinationKindSwap(newDestinationKind))
+    ).toEqual({
+      ...INITIAL_STATE,
+      destinationKind: newDestinationKind,
+      destinationKindOptions: without(ALL_CRYPTO_KIND_OPTIONS, newOriginKind),
+      destinationAmount: buildDestinationAmount(
+        INITIAL_STATE.originAmount,
+        newOriginKind,
+        newDestinationKind,
+        fakeBityRates
+      )
+    });
+  });
+
+  it('should handle SWAP_ORIGIN_AMOUNT', () => {
+    const originAmount = 2;
+    expect(
+      swap(undefined, swapActions.originAmountSwap(originAmount))
+    ).toEqual({
+      ...INITIAL_STATE,
+      originAmount
+    });
+  });
+
+  it('should handle SWAP_DESTINATION_AMOUNT', () => {
+    const destinationAmount = 2;
+    expect(
+      swap(undefined, swapActions.destinationAmountSwap(destinationAmount))
+    ).toEqual({
+      ...INITIAL_STATE,
+      destinationAmount
+    });
+  });
+
+  it('should handle SWAP_LOAD_BITY_RATES_SUCCEEDED', () => {
+    const bityRates = {
+      BTCETH: 0.01,
+      ETHREP: 10
+    };
+    expect(
+      swap(undefined, swapActions.loadBityRatesSucceededSwap(bityRates))
+    ).toEqual({
+      ...INITIAL_STATE,
+      isFetchingRates: false,
+      bityRates
+    });
+  });
+
+  it('should handle SWAP_STEP', () => {
+    const step = 2;
+    expect(swap(undefined, swapActions.changeStepSwap(step))).toEqual({
+      ...INITIAL_STATE,
+      step
+    });
+  });
+
+  it('should handle SWAP_DESTINATION_ADDRESS', () => {
+    // TODO - should reducer throw if address is not in correct format?
+    const destinationAddress = '341a0sdf83';
+    expect(
+      swap(undefined, swapActions.destinationAddressSwap(destinationAddress))
+    ).toEqual({
+      ...INITIAL_STATE,
+      destinationAddress
+    });
+  });
+
+  it('should handle SWAP_RESTART', () => {
+    // TODO - should reducer throw if address is not in correct format?
```

Comment:
IMO reducers shouldn't throw, that's on the action creator / saga to deal with. I guess just the saga, since we don't have redux-thunk or anything like that.